sql/update.go

```diff
@@ -154,6 +154,24 @@ func (p *planner) Update(n *parser.Update) (planNode, error) {
 					log.Infof("Del %q", secondaryIndexEntry.key)
 				}
 				b.Del(secondaryIndexEntry.key)
+
+				w := write{
+					values: make(writeMap),
+				}
+
+				indexDesc := &tableDesc.Indexes[i]
+				for _, columnID := range indexDesc.ColumnIDs {
+					colDesc, err := tableDesc.FindColumnByID(columnID)
+					if err != nil {
+						return nil, err
+					}
+
+					w.values[colDesc] = rowVals[colIDtoRowIndex[columnID]]
+				}
+
+				w.constraint = indexDesc
+
+				writes = append(writes, w)
```

Comment:
I think this approach is good (it generates the error messages we want) and should be submitted as-is, but all of those map allocations are going to add quite a bit of overhead to inserts/updates. I think they could be avoided by instead decoding the key of the failed operation. If a conditional put fails, the result still contains the key that was put. By decoding the key you can find out what index it is associated with, decode the column values for the index columns and generate the error message. Am I missing something that makes this infeasible? If not, add a TODO message or file another issue.